3.6/debian-9/rootfs/libmongodb.sh

```diff
@@ -579,7 +579,7 @@ mongodb_create_keyfile() {
 mongodb_is_primary_node_initiated() {
     local node="${1:?node is required}"
```

Comment:
I think it is best if we default `node` to 127.0.0.1, i.e.

```suggestion
    local node="${1:-127.0.0.1}"
```

Reply:
Can you explain the reason? I would think that you don't want the advertised hostname of the primary replica to ever be `127.0.0.1`. That could cause clients to choose `127.0.0.1` as the target hostname for the primary.

Apart from that, this `node` argument is always going to be defined because this function is only called from the `mongodb_configure_primary` function. If you follow the `node` argument up the stack, you see it is set on line 920 with `node=$(get_mongo_hostname)`, which sets it to either `$MONGODB_ADVERTISED_HOSTNAME` or `get_machine_ip`. Again, I think this is desired here.

Reply:
Oh! I see now, you're fully right as it is performing the `rs.initiate` inside this function. In this case everything LGTM.